src/base.js

```diff
@@ -13,8 +13,225 @@
+      else if (layer.source.mozRequestFullScreen)
+        layer.source.mozRequestFullScreen();
+      else if (layer.source.msRequestFullscreen)
+        layer.source.msRequestFullscreen();
+      else if (Util.isIOS()) {
+        // *sigh* Just fake it.
+        self.isPresenting = true;
+        self.fireVRDisplayPresentChange_();
+        self.beginPresent_();
+        resolve();
+      } else
+        self.waitingForPresent_ = false;
+    }
+
+    if (!self.waitingForPresent_) {
+      document.exitFullscreen();
+      reject(new Error("Unable to present."));
+    }
+  });
+};
+
+VRDisplay.prototype.exitPresent = function() {
+  var wasPresenting = this.isPresenting;
+  this.isPresenting = false;
+  this.layer_ = null;
+
+  return new Promise(function(resolve, reject) {
+    if (wasPresenting) {
+      // TODO: Spec says that exitFullscreen returns a promise, but that
+      // doesn't seem to be implemented anywhere just yet.
+      if (document.exitFullscreen)
+        document.exitFullscreen();
+      else if (document.webkitExitFullscreen)
+        document.webkitExitFullscreen();
+      else if (document.mozCancelFullScreen)
+        document.mozCancelFullScreen();
+      else if (document.msExitFullscreen)
+        document.msExitFullscreen();
+
+      resolve();
+    } else {
+      reject(new Error("Was not presenting to VRDisplay."));
+    }
+  });
+};
+
+VRDisplay.prototype.getLayers = function() {
+  if (this.layer_) {
+    return [this.layer_];
+  }
+  return null;
+};
+
+VRDisplay.prototype.fireVRDisplayPresentChange_ = function() {
+  var event = new CustomEvent('vrdisplaypresentchange', { 'vrdisplay': this });
+  window.dispatchEvent(event);
+};
+
+VRDisplay.prototype.addFullscreenListeners_ = function(element, changeHandler, errorHandler) {
+  this.removeFullscreenListeners_();
+
+  this.fullscreenEventTarget_ = element;
+  this.fullscreenChangeHandler_ = changeHandler;
+  this.fullscreenErrorHandler_ = errorHandler;
+
+  if (changeHandler) {
+    element.addEventListener("fullscreenchange", changeHandler, false);
+    element.addEventListener("webkitfullscreenchange", changeHandler, false);
+    element.addEventListener("mozfullscreenchange", changeHandler, false);
+    element.addEventListener("msfullscreenchange", changeHandler, false);
+  }
+
+  if (errorHandler) {
+    element.addEventListener("fullscreenerror", errorHandler, false);
+    element.addEventListener("webkitfullscreenerror", errorHandler, false);
+    element.addEventListener("mozfullscreenerror", errorHandler, false);
+    element.addEventListener("msfullscreenerror", errorHandler, false);
+  }
+};
+
+VRDisplay.prototype.removeFullscreenListeners_ = function() {
+  if (!this.fullscreenEventTarget_)
+    return;
+
+  var element = this.fullscreenEventTarget_;
+
+  if (this.fullscreenChangeHandler_) {
+    var changeHandler = this.fullscreenChangeHandler_;
+    element.removeEventListener("fullscreenchange", changeHandler, false);
```

Comment:
perhaps it'd be helpful to have a helper method `removeFullscreenEventListener_` that would handle all the vendor-prefixed event names?